lwm2m: Use DTLS CID by default (#13727)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

              ${COMMON_ROOT}/src/app/matter_event_handler.cpp
          )

> * For LwM2M applications, replace all use of :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` with :kconfi:option:`CONFIG_LWM2M_DTLS_CID`.

Reviewer: ```suggestion
* For LwM2M applications, replace the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option with :kconfi:option:`CONFIG_LWM2M_DTLS_CID`.
```

              ${COMMON_ROOT}/src/app/matter_event_handler.cpp
          )

> * For LwM2M applications, replace the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option with :kconfi:option:`CONFIG_LWM2M_DTLS_CID`.

Reviewer: ```suggestion
* For LwM2M applications, replace the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option with :kconfig:option:`CONFIG_LWM2M_DTLS_CID`.
```
